app/services/exports/case_log_export_service.rb

```diff
@@ -158,32 +150,62 @@ def build_manifest_xml(record_number)
       xml_doc_to_temp_file(doc)
     end
 
-    def apply_cds_transformation!(attribute_hash)
-      # OLD_CORE FORM ID support
-      attribute_hash["form"] = attribute_hash["old_form_id"]
-      attribute_hash["newform"] = attribute_hash["id"] + LOG_ID_OFFSET
+    def apply_cds_transformation(case_log)
+      attribute_hash = case_log.attributes_before_type_cast
+
+      attribute_hash["form"] = attribute_hash["old_form_id"] || (attribute_hash["id"] + LOG_ID_OFFSET)
+
+      # Changes specific to collection start year
+      if case_log.collection_start_year == 2021
+        attribute_hash.delete("joint")
+      end
+      if case_log.collection_start_year == 2022
+        attribute_hash.delete("underoccupation_benefitcap")
+      end
+
+      # Organisation fields
+      if case_log.owning_organisation.present?
+        attribute_hash["owningorgid"] = case_log.owning_organisation.old_visible_id || (case_log.owning_organisation.id + LOG_ID_OFFSET)
+        attribute_hash["owningorgname"] = case_log.owning_organisation.name
+        attribute_hash["hcnum"] = case_log.owning_organisation.housing_registration_no
+      end
+      if case_log.managing_organisation.present?
+        attribute_hash["maningorgid"] = case_log.managing_organisation.old_visible_id || (case_log.managing_organisation.id + LOG_ID_OFFSET)
+        attribute_hash["maningorgname"] = case_log.managing_organisation.name
+        attribute_hash["manhcnum"] = case_log.managing_organisation.housing_registration_no
+      end
+
+      # Mapping which would require a change in our data model
+      attribute_hash["createddate"] = attribute_hash["created_at"]
+      attribute_hash["uploaddate"] = attribute_hash["updated_at"]
+      attribute_hash["tenancycode"] = attribute_hash["tenancy_code"]
+      attribute_hash["ppcodenk"] = attribute_hash["previous_postcode_known"]
```

Comment:
Maybe this needs be `ppcodekn` oppose to `ppcodenk` ? Unless already agreed upon with the client?

Reply:
`ppcodenk` is already existing.
I think it stands for "not known" (which is confusing with ours).